WordPress/src/main/java/org/wordpress/android/ui/selfhostedusers/SelfHostedUserComposables.kt

```diff
@@ -0,0 +1,225 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import org.wordpress.android.R
+import org.wordpress.android.ui.compose.theme.M3Theme
+
+/**
+ * These composables were created for the self-hosted users feature but were written to be reusable
+ * in other projects.
+ */
+
+/**
+ * A composable that displays an avatar image optionally with a click listener. This
+ * is suitable for use in a list, such as in the self-hosted user list feature.
+ */
+@Composable
+fun SmallAvatar(
+    avatarUrl: String?,
+    onAvatarClick: ((String?) -> Unit)? = null,
+) {
+    val extraModifier = if (onAvatarClick != null) {
+        Modifier.clickable {
+            onAvatarClick(avatarUrl)
+        }
+    } else {
+        Modifier
+    }
+
+    if (avatarUrl.isNullOrEmpty()) {
+        Icon(
+            imageVector = ImageVector.vectorResource(id = R.drawable.ic_user_placeholder_primary_24),
+            contentDescription = null,
+            tint = MaterialTheme.colorScheme.onSurface,
+            modifier = Modifier
+                .size(48.dp)
+                .then(
+                    extraModifier
+                )
+        )
+    } else {
+        AsyncImage(
+            model = ImageRequest.Builder(LocalContext.current)
+                .data(avatarUrl)
+                .error(R.drawable.ic_user_placeholder_primary_24)
+                .crossfade(true)
+                .build(),
+            contentScale = ContentScale.Fit,
+            contentDescription = null,
+            modifier = Modifier
+                .clip(CircleShape)
+                .size(48.dp)
+                .then(
+                    extraModifier
+                )
+        )
+    }
+}
+
+/**
+ * A composable that displays an avatar image at the maximum screen size
+ */
+@Composable
+fun LargeAvatar(avatarUrl: String) {
+    AsyncImage(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(avatarUrl)
+            .error(R.drawable.ic_user_placeholder_primary_24)
+            .crossfade(true)
+            .build(),
+        contentScale = ContentScale.Fit,
+        contentDescription = null,
+        modifier = Modifier
+            .fillMaxSize()
+            .background(MaterialTheme.colorScheme.background)
+    )
+}
+
+/**
+ * A composable that displays a message when there is no network connection
+ */
+@Composable
+fun OfflineView() {
+    MessageView(
+        R.drawable.ic_wifi_off_24px,
+        R.string.no_network_title,
+        R.string.no_network_message,
```

Comment:
I presume this was an existing string and unrelated to the proposed changes, but I feel displaying "No network available" (title) and "There is no network available" (message) next to each other is redundant and unhelpful. 

If we wanted to improve the messaging, we might consider updating the message to say "Try again after reconnecting to the internet." However, if we do update the string, we should verify it makes sense for all current usages.

Reply:
> I presume this was an existing string and unrelated to the proposed changes, but I feel displaying "No network available" (title) and "There is no network available" (message) next to each other is redundant and unhelpful.

I agree here, but you're correct I was using existing wording used elsewhere in the app. I filed #21283 to address this.